### ISSUE TITLE
Add preview for snippets

### ISSUE THREAD

**Issue by Shatur** [hrsh7th/vim-compete#2]
I suggest adding a preview for snippets as we discussed [here](https://github.com/hrsh7th/vim-vsnip/issues/41):
![изображение](https://user-images.githubusercontent.com/22453358/77235227-b7ce5d80-6bbc-11ea-9fbe-af4f2e796c38.png)

**Comment by hrsh7th** [comment]
This feature is a bit complex to implement.

1. Porting floating-window implementation to `vim-vital-vs` from `vim-lamp`

2. Implement floating-docs feature to `vim-compete`

3. Remove floating-docs feature from `vim-lamp`

4. Modify `completionItem/resolve` feature in `vim-lamp`.

5. Providing floating-docs information to `vim-compete` from `vim-vsnip-integ`

④ needs more careful design.

I have been wanting to implement ① but another step is too heavy...

**Comment by hrsh7th** [comment]
@Shatur95
By the way, Do you know the `coc.nvim`?
It has well-supported LSP features and huge users community.

Trying to use it is good maybe if you do not know it.

**Comment by Shatur** [comment]
Yes, I know. `coc.nvim` does not support semantic highlighting, that very important for C++. It's implemented in [vim-lsp](https://github.com/prabirshrestha/vim-lsp/pull/464) and [LanguageClient-neovim](https://github.com/autozimu/LanguageClient-neovim/pull/954). Also I prefer more minimalist plugins that do one thing well.
Previously I used MUcomplete, but it was not as fast as asyncomplete.vim. So, currently I use vim-lsp + asyncomplete + vim-vsnip. Asyncomplete has some inconveniences - file and spell sources slow down editing (so, I disabled it). Compete looks like good alternative to asyncomplete (at least it have fast file completion and fast fuzzy completion). Therefore, I periodically test it to provide you some feedback (I hope it will be useful). You are actively developing it, which means that you are unsettled by current solutions.

**Comment by hrsh7th** [comment]
For tools I use often, I prefer an environment where I can change the behavior as I like.

I like the minimalist plugin too.
Thanks for explaining your opinion.

If my plugins are not enough, feel free to mention it to me.
(I will respond as far as I can)


_First sentence was translated by Google translate._
